src/issuer-helper.ts

```diff
@@ -20,28 +25,77 @@ const getExpirationDate = (credentialRecord) => {
 
 // NOTE: FEEL FREE TO ALTER IT TO CONTAIN LOGIC FOR RETRIEVING CREDENTIALS FOR LEARNERS IN YOUR ORG
 // NOTE: HOLDER ID IS GENERATED FROM AN EXTERNAL WALLET, NOT THE ISSUER
-// Method for issuer to retrieve credential on behalf of learner
-const credentialRequestHandler = async (issuerId: string, holderId: string, accessToken: string): Promise<any> => {
-  // NOTE: using one credential type for now
-  // Select credential type
-  const credentialType = 'Certificate';
-  // NOTE: CREDENTIAL ID IS THE LEARNER EMAIL IN EARLY DEPLOYMENTS OF THIS CODE
+// Method for issuer to retrieve credential on behalf of learner via OIDC token
+const processCredentialRequestViaOidc = async (issuerId: string, holderId: string, accessToken: string): Promise<any> => {
   // Select credential primary key
   const primaryKey = 'credentialSubject.email';
-  // Extract email from ID token
-  const { oidcIssuerUrl } = getConfig();
-  const oidcIssuer = await Issuer.discover(oidcIssuerUrl as string);
-  const userinfoEndpoint = oidcIssuer.metadata.userinfo_endpoint as string;
-  const userinfo = (await axios.get(userinfoEndpoint, { headers: { 'Authorization': `Bearer ${accessToken}` } })).data;
-  const email = userinfo.email;
-  if (!email) {
-    throw new Error('ID token does not contain email');
+  // Retrieve email from userinfo endpoint using OIDC token
+  let email;
+  try {
+    const { oidcIssuerUrl } = getConfig();
+    const oidcIssuer = await Issuer.discover(oidcIssuerUrl as string);
+    const userinfoEndpoint = oidcIssuer.metadata.userinfo_endpoint as string;
+    const userinfo = (await axios.get(userinfoEndpoint, { headers: { 'Authorization': `Bearer ${accessToken}` } })).data;
+    email = userinfo.email;
+    if (!email) {
+      throw new Error;
+    }
+  } catch (error) {
+    throw new Error('Could not retrieve email from userinfo endpoint using OIDC token');
+  }
+
+  let credentialRecord;
+  try {
+    const CredentialModel = await dbCredClient.open();
+    const credentialQuery = { [primaryKey]: email };
+    credentialRecord = await CredentialModel.findOne(credentialQuery);
+    await dbCredClient.close();
+  } catch (error) {
+    throw new Error('Could not retrieve credential for given email');
   }
+  if (!credentialRecord) {
+    return Promise.resolve({});
```

Comment:
```suggestion
    return {};
```
All return values from an `async` function are already wrapped in a Promise, so no need to resolve here.